MultiYearStudies.R

```diff
@@ -1,44 +1,53 @@
 
-library(skimr)
 
-#duration
-meta_df %>%
-  group_by(Study_number, OtherFactor, Ecosystem_type) %>%
-  mutate(Duration = Study_midyear - min(Study_midyear)) -> meta_time
+#conduct meta-analysis
+#this is the best fitting model of irrigiation studies - not sure if the "/" notation is correct
+#does it makes sense to have Ecosystem_type as a random effect???? Maybe???
 
 
-#Not working at the moment
-meta_time %>%
-  select(-Study_number, -Record_number, -OtherFactor, -Manipulation_level, -Soil_type) %>%
-  skim() -> summary
 
+res_irrigation <- rma(yi, vi, random = ~ 1|Study_number/Ecosystem_type,
+              mods = ~Duration*Percent_control*Latitude,
+              method = "ML",
+           data = metadat[metadat$Manipulation == "Irrigation",],
+           test = "knha")
+summary.rma(res_irrigation)
+profile(res_irrigation)
+
+
+res_drought <- rma.mv(yi, vi, random = ~ 1|Study_number/Ecosystem_type,
+                         mods = ~Duration * Percent_control * Latitude,
+                         method = "ML",
+                         data = metadat_short[metadat_short$Manipulation == "Drought",])
+summary.rma(res_drought)
+profile(res_drought)
+
+library(MuMIn)
+if (!require("devtools")) {
```

Comment:
Don't do this. (1) The script won't work without `devtools` so there's no point in using `require`. (2) Wait what? We don't want to install `dmetar` every time we run, so line 29 should be _inside_ the if block. (3) It's better to just fail with a message saying where `dmetar` can be found.

Reply:
+1 to Ben's comments, typically we only use `install.packages()` on the console line or in one of these if statements. For your purposes i would just remove the if statement and if you want to make sure users load `devtools` and `dmetar`, add it as a `library()` call

Reply:
Also here, this is a legacy of my sloppy coding, apologies! This was a dead end that was ready to be deleted (and now has been).